docs(logging): DOC-532 add page on logging (#1232)
armory/docs: content/en/armory-enterprise/armory-admin/observe/integrations-logging.md

---
title: Enable Logging in Armory Enterprise
linkTitle: Enable Logging
description: >

>    Log data about what individual accounts and functions are doing within Armory Enterprise. Push data to the your chosen aggregator. Includes how to configure Armory Enterprise to send data to Splunk and contains example dashboards.

Reviewer: ```suggestion
   Log data about what individual accounts and functions are doing within Armory Enterprise. Push data to the your chosen aggregator, includeing how to configure Armory Enterprise to send data to Splunk and contains example dashboards.
```

---

## Metrics and logging overview

Metrics allow you to investigate the health of your Armory Enterprise system. Armory's [Observability Plugin]({{< ref "observability-configure.md" >}}) provides a good overview of data over time about what is happening in particular services. You can monitor data points such as overall load, number of errors, and connection speeds.

> Logging, on the other hand, is more useful to trace data about what individual accounts and functions are doing within Armory Enterprise. This data provides an overview on applications and errors and is primarily for auditing purposes. Armory Enterprise captures logging data in Echo, so you can configure Echo to send logging data to your data aggregator.

Reviewer: ```suggestion
Logging, on the other hand, is more useful to trace data about what individual accounts and functions are doing within Armory Enterprise. This data provides an overview on applications and errors and is primarily for auditing purposes. Armory Enterprise captures logging data in the Echo service, so you can configure Echo to send logging data to your data aggregator.
```

Reviewer: Minor, but I have a sneaking suspicion a lot of ppl don't know the Spinnaker services

{{< tabs name="enable-logging" >}}
{{< tabbody name="Operator" >}}

Add the <b>rest</b> block to the <b>spec.spinnakerConfig.profiles.echo</b> section of your Operator manifest.

{{< prism lang="yaml" >}}
apiVersion: spinnaker.armory.io/v1alpha2
kind: SpinnakerService
metadata:
  name: spinnaker
spec:
  spinnakerConfig:
    profiles:
      echo:
        rest:
          enabled: true
          endpoints:
          - wrap: true
            url: "http://<HOSTNAME>:<PORT>/services/collector/event?"
            headers:
              Authorization: "<TOKEN>"
            template: '{"event":{{event}} }'
          insecure: true
{{< /prism  >}}

{{< /tabbody >}}
{{< tabbody name="Halyard" >}}

Add the <b>rest</b> block to the <b>echo-local.yml</b> section in your <b>~/.hal/default/profiles/</b> directory.

{{< prism lang="yaml" >}}
rest:
  enabled: true
  endpoints:
  - wrap: true
    url: "http://<HOSTNAME>:<PORT>/services/collector/event?"
    headers:
      Authorization: "<TOKEN>"
    template: '{"event":{{event}} }'
  insecure: true
{{< /prism  >}}

{{< /tabbody >}}
{{< /tabs >}}

> The `HOSTNAME` and `PORT` should be the same as the TCP listener port available in your aggregator. `TOKEN` is the authorization token for your aggregator.

Reviewer 2: Only the PORT is configured by the TCP listener port config. The HOSTNAME is the FQDN of the service i.e. https://splunk.armory.io. An IP:PORT can also work in some scenarios so maybe calling it HOSTNAME is a little misleading carry over from the old docs and we should update it to something else. SERVICE_URL maybe?

> The `HOSTNAME` and `PORT` should be the same as the TCP listener port available in your aggregator. `TOKEN` is the authorization token for your aggregator.

Reviewer 2: The default port for HEC is either 8088 or 8089 depending on your Splunk version. This port will be used if you DO NOT configure a TCP listener. Any TCP listener config is a custom port assignment.

Reviewer 3: With regards to Chris' findings, I placed the following KB article about the ports, and it can be linked to the article: https://support.armory.io/support?id=kb_article_view&sysparm_article=KB0010554

> The `HOSTNAME:PORT` can be HTTPS, which would remove the need for the insecure TOKEN. You should consult your specific data aggregator provider's documentation for how to configure an HTTPS endpoint.

Reviewer: ```suggestion
The `HOSTNAME:PORT` can be HTTPS, which removes the need for the insecure `TOKEN`. You should consult your specific data aggregator provider's documentation for how to configure an HTTPS endpoint.
```

## Verify data flow to your aggregator

> You can either wait for events to flow or run a new pipeline to test that Echo is sending logging details to your aggregator. You should now be able to search the Armory Enterprise index and see events coming in. If you don't see any data, Echo's logs should point to the source of the problem.

Reviewer: ```suggestion
You can either wait for events to flow or run a new pipeline to test that Echo is sending logging details to your aggregator. You should now be able to search the Armory Enterprise index and see events coming in. If you don't see any data, Echo's logs may point to the source of the problem.
```
